src/hotspot/share/compiler/compilerOracle.cpp

```diff
@@ -334,6 +336,16 @@ static void register_command(TypedMethodOptionMatcher* matcher,
     return;
   }
 
+  if (!UnlockDiagnosticVMOptions) {
+    const char* name = option2name(option);
+    JVMFlag* flag = JVMFlag::find_declared_flag(name);
+    if (flag != nullptr && flag->is_diagnostic()) {
+      jio_snprintf(errorbuf, buf_size, "VM option '%s' is diagnostic and must be enabled via -XX:+UnlockDiagnosticVMOptions.", name);
```

Comment:
Just noticed that code above in line 333 uses `warning(...)`. I think that code should also use `errorbuf` now that we have it.

Reply:
Ah right, I missed that. Let's revert this part of the change then.

Reply:
Reverted